### PR TITLE
Revisions

### PR DESCRIPTION
**Pull request by d00rman**
This PR is a continuation of #168 so please make sure to merge that one before considering this.

It adds the /revision/ and /revision/{revision} endpoints to the page_revisions module and exposes them in the public API.

### CODE REVIEW

File: mods/page_revisions.js [hunk 1]
```diff
@@ -223,21 +237,68 @@ PRS.prototype.listTitleRevisions = function(restbase, req) {
...
+        // ask the MW API directly and
+        // store and return its result
+        return this.fetchAndStoreMWRevision(restbase, req);
+    }
+    // check the storage, and, if no match is found
+    // ask the MW API about the revision
+    return restbase.get({
+        uri: this.tableURI(rp.domain),
+        body: {
+            table: this.tableName,
```
Comment (gwicke) [inline mods/page_revisions.js]: add `index: 'by_rev'`

File: mods/page_revisions.js [hunk 2]
```diff
@@ -68,6 +70,7 @@ PRS.prototype.getTableSchema = function () {
                 { attribute: 'rev', type: 'hash' },
                 { attribute: 'tid', type: 'range', order: 'desc' },
                 { attribute: 'title', type: 'range', order: 'asc' },
+                { attribute: 'hidden', type: 'proj' },
                 { attribute: 'tags', type: 'proj' }
```
Comment (gwicke) [inline mods/page_revisions.js]: We can probably drop tags from the projection if we are using 'hidden' for acls.

File: mods/page_revisions.js [hunk 3]
```diff
@@ -43,8 +43,10 @@ PRS.prototype.getTableSchema = function () {
             rev: 'int',             // MediaWiki oldid
             latest_rev: 'int',      // Latest MediaWiki revision
             tid: 'timeuuid',
+            // revision deletion or suppression, can be:
+            // - sha1hidden, commenthidden, texthidden
+            hidden: 'set<string>',
```
Comment (gwicke) [inline mods/page_revisions.js]: Alternative names:
- restrictions
- acls

File: mods/page_revisions.js (hunk 2 above)
Comment (d00rman) [inline mods/page_revisions.js]: excellent idea!

**Comment by gwicke** [comment]
@d00rman, @earldouglas : LGTM, but leaving the merge to James as I contributed a couple of patches myself.

**Comment by d00rman** [comment]
I have added one more [revision test](https://github.com/wikimedia/restbase/commit/d5385775a4d8a65914f65e6bd9abd4d26bc71708) and [moved some code around](https://github.com/wikimedia/restbase/commit/741afa9345869dc54ad1b5ad22fdc8ad1fb721e5) in tests - extracted `localRequests()` and the like from the Parsoid tests and put them in `assert.js`. @earldouglas please confirm that's OK.

**Comment by gwicke** [comment]
Merged, as the Travis failure was a benign 'coveralls upload failed' error. The env var sometimes isn't set in travis it seems.

**Comment by gwicke** [comment]
TODO: Add a test for listRevisions.

**Comment by d00rman** [comment]
@gwicke yep, I know, have it on my in-mind TODO list. Haven't put it in this branch/PR to avoid conflicts with #169 . Will put it, though

**Comment by d00rman** [comment]
@gwicke Addressed in #172 in 5bbc4a7d124bab88f9f3bc50bdc3e861df244303

**Pull request merged**